### ISSUE TITLE
[BUG] The heat map resolution is too low

### ISSUE THREAD

**Issue by ltzg** [jamesdolezal/slideflow#279]
I used these code to creaate a heatmap for SVS pitures in TCGA. However, the heat map resolution is too low.
###############
P.extract_tiles(tile_px=512, tile_um=256)

from slideflow.model import build_feature_extractor

resnet50 = build_feature_extractor(
    'resnet50_imagenet',
    tile_px=512
)

dataset = P.dataset(tile_px=512, tile_um=256)

# Calculate features for this dataset.
features = sf.DatasetFeatures(resnet50, dataset)

Exporting Features
features.to_torch('./data/features')###导出特征文件
Model Configuration
import slideflow as sf
from slideflow.mil import mil_config
config = mil_config('TransMIL', lr=1e-3)Training an MIL Model
# Prepare a project and dataset

full_dataset = dataset
# Split the dataset using three-fold, site-preserved cross-validation
splits = full_dataset.kfold_split(
    k=3,
    labels='category'
)

# Train on each cross-fold
for train, val in splits:
    P.train_mil(
        config=config,
        outcomes='category',
        train_dataset=train,
        val_dataset=val,
        bags='./data/features'
    )##点两次回车键

# Generate attention heatmaps,# using the 'magma' colormap and no interpolation.
for train, val in splits:
    P.train_mil(
        config=config,
        outcomes='category',
        train_dataset=train,
        val_dataset=val,
        bags='./data/features',
        interpolation=None,
        attention_heatmaps=True,
        cmap='magma'
)
#################
![image](https://github.com/jamesdolezal/slideflow/assets/51331990/ab3cfa7f-b160-475c-9ee2-85f0aeb16f8a)

**Comment by ltzg** [comment]
Dear Dolezal,Thank you very much for your letter. The heat map I made is shown below. There are a lot of Mosaic points in the picture (up). However, Other researchers have made very smooth heat maps(down). How am I supposed to do that. 
    
  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            James ***@***.***>
            
        
      
      
        
         Date 
        
        
        5/19/2023 23:37
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [jamesdolezal/slideflow] [BUG] The heat map resolution is too low (Issue #279)
        
      
    
  

Hi Itzg - thanks for creating an issue. Can you clarify what the issue is? The heatmap resolution appears appropriate at first glance.

**Comment by jamesdolezal** [comment]
The resolution you're seeing is a reflection of the tile size - each image tile gets one associated attention value. You can increase the "resolution" of the heat map either by 1) training models at higher resolution (40x instead of 20x), or 2) extracting tiles with overlapping stride.

To accomplish the first, replace `P.extract_tiles(tile_px=512, tile_um=256)` with `P.extract_tiles(tile_px=512, tile_um='40x')` or `tile_um=128`.

To accomplish the second, set `stride_div` equal to a positive number, which will extract the tiles with overlap. To extract tiles with 4-fold overlap (which will generate `4^2`, or 16-times more data), do `P.extract_tiles(..., stride_div=4)`. This will greatly increasing your dataset size, but will increase the apparent resolution of the attention heat maps.
